search.dart

```diff
@@ -0,0 +1,146 @@
+    required this.url,
+    required this.category,
+  });
+}
+//I am unsure about what needs to happen here: Import the data maybe?
+const List<Video> allVideos = [
+  Video(
+    title: 'Breathing exercise ',
+    url: 'https://thc.com/breathing.mp4',
+    category: 'Breathing',
+  ),
+  Video(
+    title: 'Video 2',
+    url: 'https:..mp4',
+    category: 'Meditation',
+  ),
+  // Do we have a databases for video's /how will this work?
+];
+
+class MainPage extends StatefulWidget {
+  @override
+  _MainPageState createState() => _MainPageState();
+}
+
+class _MainPageState extends State<MainPage> {
+  final TextEditingController controller = TextEditingController();
+  String selectedCategory = 'All';
+  List<Video> videos = allVideos;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Search and Filter Videos'),
+      ),
+      body: Column(
+        children: <Widget>[
+          Container(
+            margin: const EdgeInsets.fromLTRB(16, 16, 16, 0),
+            child: TextField(
+              controller: controller,
+              decoration: InputDecoration(
+                prefixIcon: const Icon(Icons.search),
+                hintText: 'Search here',
+                border: OutlineInputBorder(
+                  borderRadius: BorderRadius.circular(8),
+                  borderSide: const BorderSide(),
+                ),
+              ),
+              onChanged: searchVideo,
+            ),
+          ),
+          DropdownButton<String>(
+            value: selectedCategory,
+            onChanged: (newValue) {
+              setState(() {
+                selectedCategory = newValue!;
+                filterVideos();
+              });
+            },
+            items: _buildCategoryDropdownItems(),
+          ),
+          Expanded(
+            child: ListView.builder(
+              itemCount: videos.length,
+              itemBuilder: (context, index) {
+                final video = videos[index];
+                return ListTile(
+                  leading: Image.network(
+                    video.url,
+                    fit: BoxFit.cover,
+                    width: 50,
+                    height: 50,
+                  ),
+                  title: Text(video.title),
+                  subtitle: Text(video.category),
+                  onTap: () => Navigator.push(
+                    context,
+                    MaterialPageRoute(builder: (context) => VideoPage(video: video)),
+                  ),
+                );
+              },
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  List<DropdownMenuItem<String>> _buildCategoryDropdownItems() {
+    List<DropdownMenuItem<String>> items = ['All'] + allVideos.map((video) => video.category).toSet().toList()
+      ..sort()
+      .map((category) => DropdownMenuItem(
+            value: category,
+            child: Text(category),
+          ))
+      .toList();
```

Comment:
You've mastered the art of `List.map` and the `..sort()` cascade syntax. I think you're ready for a new technique, which will make your Dart coding even more elegant: [**control flow collection operators**](https://dart.dev/language/collections#control-flow-operators).

```suggestion
    final categories = {
      'All',
      for (final video in allVideos) video.category,
    };
    final items = [
      for (final category in categories.toList()..sort())
        DropdownMenuItem<String>(value: category, child: Text(category)),
    ];
```